Fix #252: lookForAtArea matrix result type.

zyzyzyryxy opened the pull request:
### Brief Description

Fix incorrect typing for return value of room.lookForAtArea matrix form (isArray = false)

### Checklists

- [x] Test passed
- [x] Coding style (indentation, etc) - hope so, I let workpace-defined formatter do its job
- [x] Edits have been made to `src/` files not `index.d.ts`
- [x] Run `npm run dtslint` to update `index.d.ts`

Jomik commented:
We are missing tests for this change, it is a bit hard to see what this actually accomplishes in the context of this PR.

zyzyzyryxy commented:
> We are missing tests for this change, it is a bit hard to see what this actually accomplishes in the context of this PR.

I thought I did add the tests, but maybe I am missing something about the project structure? Thought dist/screeps-tests.ts contains tests, I added lines 919 - 921 with a test from the issue reported and fixed previous test - line 911.

Jomik commented:
> > We are missing tests for this change, it is a bit hard to see what this actually accomplishes in the context of this PR.
> 
> 
> 
> I thought I did add the tests, but maybe I am missing something about the project structure? Thought dist/screeps-tests.ts contains tests, I added lines 919 - 921 with a test from the issue reported and fixed previous test - line 911.

You did, I just didn't have coffee yet 😅 
Maybe add the same with an explicit `false` for the `asArray` argument.
Then this looks good. Amazed that this has been an error for so long  .

zyzyzyryxy commented:
> > > We are missing tests for this change, it is a bit hard to see what this actually accomplishes in the context of this PR.
> > 
> > 
> > I thought I did add the tests, but maybe I am missing something about the project structure? Thought dist/screeps-tests.ts contains tests, I added lines 919 - 921 with a test from the issue reported and fixed previous test - line 911.
> 
> You did, I just didn't have coffee yet 😅 Maybe add the same with an explicit `false` for the `asArray` argument. Then this looks good. Amazed that this has been an error for so long .

Done

DiamondMofeng reviewed (approved):
Thanks for fixing this issue
